xmppserver/src/main/java/org/jivesoftware/openfire/pep/IQPEPHandler.java

```diff
@@ -428,6 +385,31 @@ else if (packet.getType() == IQ.Type.get || packet.getType() == IQ.Type.set) {
         return null;
     }
 
+    /**
+     * Populates the PEPService instance with subscriptions. The subscriptions that
+     * are added to the PEPService are based on the roster of the owner of the PEPService:
+     * every entity that's subscribed to the presence of the owner, is added as a
+     * subscriber of the PEPService.
+     *
+     * This method adds, but does not remove of update existing subscriptions.
+     *
+     * @param pepService The PEPService to be populated with subscriptions.
+     */
+    public void addSubscriptionForRosterItems( final PEPService pepService )
+    {
+        try {
+            final RosterManager rm = XMPPServer.getInstance().getRosterManager();
+            final Roster roster = rm.getRoster(pepService.getAddress().getNode());
+            for (final RosterItem item : roster.getRosterItems()) {
+                if (item.getSubStatus() == RosterItem.SUB_BOTH || item.getSubStatus() == RosterItem.SUB_FROM) {
+                    createSubscriptionToPEPService(pepService, item.getJid(), pepService.getAddress());
+                }
```

Comment:
What's the `else` block? If it should never happen, throw an IllegalStateException. If it's deliverately ignore, add a log statement so at least during debugging it's possible to figure it out.

Reply:
I think this is quite correct as-is? PEP subscription is based on presence subscription, so when a new PEP service is created, it's subscriptions should be based on the subscriptions in the roster. The states that are evaluated here are those that reflect an entity being subscribed to the owner-of-the-PEP-service presence (the others are not). Adding a log statement for those might confuse people more than that it prevents confusion, I'd think.